### PR TITLE
Simplified transaction DTOs usage

### PR DESCRIPTION
**Pull request by iulianpascalau**
- simplified transaction DTOs usage
- linter issue fixes

### CODE REVIEW

File: builders/relayedTxV1Builder.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ func prepareInnerTxForRelayV1(tx *data.Transaction) (string, error) {
 		return "", err
 	}
 
+	// TODO: remove this hardcoded implementation in core. Create there a dedicated converter between FrontendTransaction <-> Transaction
```
Comment (bogdan-rosianu) [inline builders/relayedTxV1Builder.go:115]: // TODO: remove this hardcoded implementation. Inside mx-chain-core-go, create there a dedicated converter between FrontendTransaction <-> Transaction

Comment (dragos-rebegea) [inline builders/relayedTxV1Builder.go:115]: :+1:

File: examples/examplesTransaction/main.go [hunk 2]
```diff
@@ -83,25 +83,25 @@ func main() {
 		return
 	}
 
-	tx, err := ti.ApplySignatureAndGenerateTx(holder, transactionArguments)
+	err = ti.ApplySignature(holder, &tx)
 	if err != nil {
 		log.Error("error creating transaction", "error", err)
```
Comment (sstanculeanu) [inline examples/examplesTransaction/main.go:88]:
```suggestion
		log.Error("error signing transaction", "error", err)
```

**Review by dragos-rebegea — APPROVED** [review]

File: builders/txBuilder.go [hunk 3]
```diff
@@ -32,46 +31,28 @@ func NewTxBuilder(signer Signer) (*txBuilder, error) {
...
 
-	return builder.createTransaction(arg), nil
+	return nil
 }
 
 // ComputeTxHash will return the hash of the provided transaction. It assumes that the transaction is already signed,
 // otherwise it will return an error.
 // The input can be the result of the ApplySignatureAndGenerateTx function
-func (builder *txBuilder) ComputeTxHash(tx *data.Transaction) ([]byte, error) {
+func (builder *txBuilder) ComputeTxHash(tx *transaction.FrontendTransaction) ([]byte, error) {
```
Comment (sstanculeanu) [inline builders/txBuilder.go:55]: the last line of the comment is not valid anymore

File: interactors/nonceHandlerV2/nonceTransactionsHandler.go [hunk 4]
```diff
@@ -69,20 +70,20 @@ func NewNonceTransactionHandlerV2(args ArgsNonceTransactionsHandlerV2) (*nonceTr
 }
 
 // ApplyNonceAndGasPrice will apply the nonce to the given ArgCreateTransaction
-func (nth *nonceTransactionsHandlerV2) ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, txArgs *data.ArgCreateTransaction) error {
+func (nth *nonceTransactionsHandlerV2) ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, tx *transaction.FrontendTransaction) error {
```
Comment (sstanculeanu) [inline interactors/nonceHandlerV2/nonceTransactionsHandler.go:73]: comment needs to be updated

File: interactors/interface.go [hunk 5]
```diff
@@ -12,39 +13,39 @@ import (
...
 	ForceNonceReFetch(address core.AddressHandler) error
 	Close() error
 	IsInterfaceNil() bool
 }
 
 // TransactionNonceHandlerV2 defines the component able to apply nonce for a given ArgCreateTransaction
 type TransactionNonceHandlerV2 interface {
-	ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, txArgs *data.ArgCreateTransaction) error
-	SendTransaction(ctx context.Context, tx *data.Transaction) (string, error)
+	ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, tx *transaction.FrontendTransaction) error
```
Comment (sstanculeanu) [inline interactors/interface.go:47]: interface comment needs to be updated

File: aggregator/notifees/interface.go [hunk 6]
```diff
@@ -18,14 +19,14 @@ type TxBuilder interface {
...
+	SendTransaction(ctx context.Context, tx *transaction.FrontendTransaction) (string, error)
+	SendTransactions(ctx context.Context, txs []*transaction.FrontendTransaction) ([]string, error)
 	IsInterfaceNil() bool
 }
 
 // TransactionNonceHandler defines the component able to apply nonce for a given ArgCreateTransaction
 type TransactionNonceHandler interface {
-	ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, txArgs *data.ArgCreateTransaction) error
-	SendTransaction(ctx context.Context, tx *data.Transaction) (string, error)
+	ApplyNonceAndGasPrice(ctx context.Context, address core.AddressHandler, tx *transaction.FrontendTransaction) error
```
Comment (sstanculeanu) [inline aggregator/notifees/interface.go:29]: interface comment needs to be updated

File: interactors/nonceHandlerV2/addressNonceHandler.go [hunk 7]
```diff
@@ -45,37 +45,37 @@ func NewAddressNonceHandler(proxy interactors.Proxy, address erdgoCore.AddressHa
...
 		proxy:        proxy,
-		transactions: make(map[uint64]*data.Transaction),
+		transactions: make(map[uint64]*transaction.FrontendTransaction),
 	}, nil
 }
 
 // ApplyNonceAndGasPrice will apply the computed nonce to the given ArgCreateTransaction
-func (anh *addressNonceHandler) ApplyNonceAndGasPrice(ctx context.Context, txArgs *data.ArgCreateTransaction) error {
-	oldTx, alreadyExists := anh.isTxAlreadySent(txArgs)
+func (anh *addressNonceHandler) ApplyNonceAndGasPrice(ctx context.Context, tx *transaction.FrontendTransaction) error {
```
Comment (sstanculeanu) [inline interactors/nonceHandlerV2/addressNonceHandler.go:53]: comment needs to be updated

File: interactors/nonceHandlerV2/singleTransactionAddressNonceHandler.go [hunk 8]
```diff
@@ -35,15 +35,15 @@ func NewSingleTransactionAddressNonceHandler(proxy interactors.Proxy, address er
 }
 
 // ApplyNonceAndGasPrice will apply the computed nonce to the given ArgCreateTransaction
-func (anh *singleTransactionAddressNonceHandler) ApplyNonceAndGasPrice(ctx context.Context, txArgs *data.ArgCreateTransaction) error {
+func (anh *singleTransactionAddressNonceHandler) ApplyNonceAndGasPrice(ctx context.Context, tx *transaction.FrontendTransaction) error {
```
Comment (sstanculeanu) [inline interactors/nonceHandlerV2/singleTransactionAddressNonceHandler.go:38]: comment needs to be updated

File: builders/relayedTxV1Builder.go (hunk 1 above)
Comment (iulianpascalau) [reply builders/relayedTxV1Builder.go:115]: done

File: builders/txBuilder.go [hunk 9]
```diff
@@ -126,11 +107,11 @@ func transactionToNodeTransaction(tx *data.Transaction) (*transaction.Transactio
 	}, nil
 }
 
-func (builder *txBuilder) createUnsignedTx(arg data.ArgCreateTransaction) *data.Transaction {
-	arg.Signature = ""
-	tx := builder.createTransaction(arg)
+func (builder *txBuilder) createUnsignedTx(tx transaction.FrontendTransaction) *transaction.FrontendTransaction {
+	copiedTransaction := tx
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply builders/txBuilder.go:111]: indeed. Now we do only one copy of the struct

File: examples/examplesTransaction/main.go (hunk 2 above)
Comment (iulianpascalau) [reply examples/examplesTransaction/main.go:88]: fixed

File: builders/txBuilder.go (hunk 3 above)
Comment (iulianpascalau) [reply builders/txBuilder.go:55]: removed the last line as it is irrelevant now.

File: interactors/nonceHandlerV2/nonceTransactionsHandler.go (hunk 4 above)
Comment (iulianpascalau) [reply interactors/nonceHandlerV2/nonceTransactionsHandler.go:73]: updated

File: interactors/interface.go (hunk 5 above)
Comment (iulianpascalau) [reply interactors/interface.go:47]: fixed

File: aggregator/notifees/interface.go (hunk 6 above)
Comment (iulianpascalau) [reply aggregator/notifees/interface.go:29]: done

File: interactors/nonceHandlerV2/addressNonceHandler.go (hunk 7 above)
Comment (iulianpascalau) [reply interactors/nonceHandlerV2/addressNonceHandler.go:53]: done

File: interactors/nonceHandlerV2/singleTransactionAddressNonceHandler.go (hunk 8 above)
Comment (iulianpascalau) [reply interactors/nonceHandlerV2/singleTransactionAddressNonceHandler.go:38]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**